tests/test_utility.py

```diff
@@ -158,3 +159,68 @@ def test_only_annotation_changes(example_new_dict):
         )
         is False
     )
+
+
+@pytest.mark.parametrize(
+    "gh_username, expected_prefix",
+    [("testuser", "testuser/update-"), (None, "update-")],
+)
+def test_create_random_branch_name(gh_username, expected_prefix):
+    branch_name = utils.create_random_branch_name(gh_username)
+    assert branch_name.startswith(expected_prefix)
+
+
+def test_create_commit_message():
+    test_contributor = Contributor(
+        name="Test User",
+        email="testuser@gmail.com",
+        changes_summary="Test changes",
+    )
+    assert (
+        utils.create_commit_message(test_contributor, "Add participants.json")
+        == "[bot] Add participants.json\n\nCo-authored-by: Test User <testuser@gmail.com>"
+    )
+
+
+@pytest.mark.parametrize(
+    "input_str, expected_output",
+    [
+        ("added annotations for:\\n- age", "added annotations for:\n- age"),
+        ("added annotations for age", "added annotations for age"),
+    ],
+)
+def convert_literal_newlines(input_str, expected_output):
+    assert utils.convert_literal_newlines(input_str) == expected_output
+
+
+@pytest.mark.parametrize(
+    "name, gh_username, affiliation, expected_name_in_pr",
+    [
+        ("John Doe", "johndoe", "Test University", "John Doe (@johndoe)"),
+        ("Jane Doe", None, None, "Jane Doe"),
+    ],
+)
+def test_create_pull_request_body(
+    name, gh_username, affiliation, expected_name_in_pr
+):
+    test_contributor = Contributor(
+        name=name,
+        email="user@gmail.com",
+        changes_summary="A more detailed change",
+        gh_username=gh_username,
+        affiliation=affiliation,
+    )
+    pr_body = utils.create_pull_request_body(
+        test_contributor, "Update participants.json"
+    )
+    expected_pr_body = f"""### Overview of proposed changes (bot-generated):
+Update participants.json
+
+### More details:
+A more detailed change
+
+### Changes proposed by:
+Name: {expected_name_in_pr}
+Affiliation: {affiliation}"""
```

Comment:
:cherries: I think something like https://docs.python.org/3/library/textwrap.html#textwrap.dedent would read a little nicer here

Reply:
Thanks for the suggestion! So I tried this out, and I think textwrap.dedent is actually harder to work with in this use case because the f-string includes user input (e.g. summary of changes) which may have newlines in them. `dedent` doesn’t really work with `\n` characters or inconsistent whitespace across lines, and so the final indentation in a given PR body becomes hard to predict without additional code to handle edge cases.

In this case, I think the explicit syntax may be a better choice to ensure predictable result strings.